lib/rules/declaration-property-unit-allowed-list/__tests__/index.js

```diff
@@ -175,3 +179,58 @@ testRule({
 		},
 	],
 });
+
+testRule({
+	ruleName,
+
+	config: [
+		{
+			'/^border/': ['px'],
+			'/^background/': ['%'],
+		},
+		{
+			ignore: ['inside-function'],
+		},
+	],
+
+	skipBasicChecks: true,
+
+	accept: [
+		{
+			code: 'button { border-color: hsla(162deg, 51%, 35%, 0.8); }',
+		},
+		{
+			code: 'button { border: 1px solid hsla(162deg, 51%, 35%, 0.8); }',
+		},
+		{
+			code:
+				'button { background-image: linear-gradient(hsla(162deg, 51%, 35%, 0.8), hsla(62deg, 51%, 35%, 0.8)); }',
+		},
+		{
+			code: 'button { background-image: url("https://example.com/img.png"); }',
+		},
+		{
+			code:
+				'button { background: center center / 50% 50% linear-gradient(hsla(162deg, 51%, 35%, 0.8), hsla(62deg, 51%, 35%, 0.8)); }',
+		},
+		{
+			code: 'button { background-color: var(--bg-color); }',
+		},
+	],
+
+	reject: [
+		{
+			code: 'button { border: 1rem solid hsla(162deg, 51%, 35%, 0.8); }',
+			message: messages.rejected('border', 'rem'),
+		},
+		{
+			code: 'button { background-position: 5rem 45%; }',
+			message: messages.rejected('background-position', 'rem'),
+		},
+		{
+			code:
+				'button { background: center center / 50px 10% linear-gradient(hsla(162deg, 51%, 35%, 0.8), hsla(62deg, 51%, 35%, 0.8)); }',
```

Comment:
[suggestion] We uses usually the `a` type selector on the rule README, so how about using `a` also on the test code?

```suggestion
			code: 'a { border-color: hsla(162deg, 51%, 35%, 0.8); }',
		},
		{
			code: 'a { border: 1px solid hsla(162deg, 51%, 35%, 0.8); }',
		},
		{
			code:
				'a { background-image: linear-gradient(hsla(162deg, 51%, 35%, 0.8), hsla(62deg, 51%, 35%, 0.8)); }',
		},
		{
			code: 'a { background-image: url("https://example.com/img.png"); }',
		},
		{
			code:
				'a { background: center center / 50% 50% linear-gradient(hsla(162deg, 51%, 35%, 0.8), hsla(62deg, 51%, 35%, 0.8)); }',
		},
		{
			code: 'a { background-color: var(--bg-color); }',
		},
	],

	reject: [
		{
			code: 'a { border: 1rem solid hsla(162deg, 51%, 35%, 0.8); }',
			message: messages.rejected('border', 'rem'),
		},
		{
			code: 'a { background-position: 5rem 45%; }',
			message: messages.rejected('background-position', 'rem'),
		},
		{
			code:
				'a { background: center center / 50px 10% linear-gradient(hsla(162deg, 51%, 35%, 0.8), hsla(62deg, 51%, 35%, 0.8)); }',
```